src/pages/workspaces/workspace/workflows/DataStepContent.js

```diff
@@ -86,69 +94,117 @@ export default class DataStepContent extends Component {
+                name: 'choose-set-components',
+                checked: isProcessAllSetComponents,
+                onChange: () => this.setNewSelectionModel({ type: processAllSetComponents, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            hasEntityType && div([
+              h(RadioButton, {
+                text: 'Choose existing sets',
+                name: 'choose-set-components',
+                checked: isChooseSets,
+                onChange: () => this.setNewSelectionModel({ type: chooseSets, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ])
+          ]
+          ) :
+          h(Fragment, [
+            div([
+              h(RadioButton, {
+                text: `Process all ${rootEntityTypeCount} rows`,
+                name: 'process-rows',
+                checked: isProcessAll,
+                onChange: () => this.setNewSelectionModel({ type: processAll, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            div([
+              h(RadioButton, {
+                text: 'Choose specific rows to process',
+                name: 'process-rows',
+                checked: isChooseRows,
+                onChange: () => this.setNewSelectionModel({ type: chooseRows, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            hasSet && div([
+              h(RadioButton, {
+                text: 'Choose existing sets',
+                name: 'process-rows',
+                checked: isProcessMergedSet,
+                onChange: () => this.setNewSelectionModel({ type: processMergedSet, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ])
+          ])]),
+        !isProcessAll && !isProcessAllSetComponents && div({
           style: {
             display: 'flex', flexDirection: 'column',
             height: 500, marginTop: '1rem'
           }
         }, [
           h(DataTable, {
             key: type.description,
-            entityType: isProcessMergedSet ? setType : rootEntityType,
+            childrenBefore: () => div({ style: Style.elements.sectionHeader }, [
+              Utils.cond(
+                [isProcessMergedSet, () => 'Select one or more sets to combine and process'],
+                [isChooseRows, () => `Select ${rootEntityType}s to process`],
+                [isChooseSetComponents, () => `Select ${rootEntityType.slice(0, -4)}s to create a new set to process`],
+                [isChooseSets, () => `Select up to 10 ${rootEntityType}s to process in parallel`]
+              )
+            ]),
+            entityType: isChooseSetComponents ? baseEntityType : isProcessMergedSet ? setType : rootEntityType,
```

Comment:
I'm torn, do you think this is clearer?
```suggestion
            entityType: (isChooseSetComponents && baseEntityType) || (isProcessMergedSet && setType) || rootEntityType,
```
Or should we just put a `cond` here? I'm just looking for clarity for unfamiliar readers.

Reply:
We accepted your suggestion, that works fine. Thanks!